src/mon/PGMap.cc

```diff
@@ -2056,7 +2056,8 @@ void PGMap::get_filtered_pg_stats(uint64_t state, int64_t poolid, int64_t osdid,
       continue;
     if ((osdid >= 0) && !(i->second.is_acting_osd(osdid,primary)))
       continue;
-    if (state != (uint64_t)-1 && !(i->second.state & state))
+    if (state != (uint64_t)-1 && !(i->second.state & state) &&
+	!(state == 0 && i->second.state == 0))
```

Comment:
Maybe doing it this way is clearer.  So inside the !( ) is all the ways we have a matching pg.

if (!(state == (uint64_t)-1 || i->second.state & state ||
     (state == 0 && i->second.state == 0)))

Reply:
:+1: